pkg/controllers/monitoring-stack/controller.go

```diff
@@ -130,20 +131,74 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, nil
 	}
 
+	availableCondition := metav1.Condition{
+		Type:               "Available",
+		ObservedGeneration: ms.ObjectMeta.Generation,
+		LastTransitionTime: metav1.Time{Time: time.Now().UTC()},
+	}
+
 	reconcilers := stackComponentReconcilers(ms, r.instanceSelectorKey, r.instanceSelectorValue)
 	for _, reconciler := range reconcilers {
 		err := reconciler(ctx, r.k8sClient, r.scheme)
 		// handle create / update errors that can happen due to a stale cache by
 		// retrying after some time.
 		if errors.IsAlreadyExists(err) || errors.IsConflict(err) {
 			logger.V(3).Info("skipping reconcile error", "err", err)
+			availableCondition.Status = metav1.ConditionFalse
+			availableCondition.Reason = "Failure"
+			availableCondition.Message = fmt.Sprintf("Monitoring stack failed to reconcile: %v", err)
+			err := r.updateStatusCondition(ctx, ms, availableCondition)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
 			return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
 		}
 		if err != nil {
+			availableCondition.Status = metav1.ConditionFalse
+			availableCondition.Reason = "Failure"
+			availableCondition.Message = fmt.Sprintf("Monitoring stack failed to reconcile: %v", err)
+			err := r.updateStatusCondition(ctx, ms, availableCondition)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
 			return ctrl.Result{}, err
 		}
 	}
-	return ctrl.Result{}, nil
+	availableCondition.Status = metav1.ConditionTrue
+	availableCondition.Reason = "Ready"
+	availableCondition.Message = "Monitoring stack is up and running"
+	err = r.updateStatusCondition(ctx, ms, availableCondition)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+}
+
+func (r *reconciler) updateStatusCondition(ctx context.Context, ms *stack.MonitoringStack, c metav1.Condition) error {
+	updatedMS := ms.DeepCopy()
+	newStatus := stack.MonitoringStackStatus{}
+
+	// if the condition exists and the reason hasn't changed then do no update
+	if existingC := getConditionByType(c.Type, updatedMS.Status.Conditions); existingC != nil {
+		if existingC.Reason == c.Reason {
+			return nil
+		}
+	}
+	updatedMS.Status.Conditions = append(newStatus.Conditions, c)
```

Comment:
won't this clear other conditions say `Progressing | Ready` that are set?

Reply:
yes, but there are no other conditions at the moment

Reply:
In that case do even need to search and append? IMHO, we should try and follow the implementation in Prometheus Operator.

Reply:
I guess it depends if there's only one place in the code where we update the conditions or if can done from different places. But I suppose that for now it doesn't matter much.

Reply:
> Do we just want to replicate the status conditions from the Prometheus (if so then what about the alertmanager status?) or do we want to somehow "wrap" them (i.e introduce something like `PrometheusAvailable`, `PrometheusReconciled`)?

I think its fine to simply pull in or forward the `Prometheus` CR status into the `MonitoringStack` CR status. It seems natural and ideal the `Prometheus` status already adheres to [best practices](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties).
This can also be the start of the status reporting. If we later find we want to add a condition for the operator itself or some other component that managed through a `MonitoringStack` CR, we'll simply add it.
So with that it probably makes sense to introduce our own condition type and embed the [PrometheusStatus](https://github.com/prometheus-operator/prometheus-operator/blob/89d905f764c81d3e55342fbd2919fcdba881c609/pkg/apis/monitoring/v1/types.go#L567) type. Then we can our own condition type (or others) later if needed.

I hope this makes sense, I'm far from an expert in reporting CR status.

Reply:
Thanks. I updated the PR. Please check